addon/services/horizon.js

```diff
@@ -110,11 +162,69 @@ export default Ember.Service.extend({
   },
 
   findAll(collection, filterBy) {
+    console.log('findAll', collection, filterBy);
+  },
 
+  subscribe(input) {
+    return collection.subscribe(input);
   },
 
+  /**
+   * Called whenever a change is detected by one of the watched
+   * collections. It then calls any registered callbacks that have
+   * expressed interest via the "registerCallback" function
+   *
+   * @param  {hash} meta includes "changes", "connection", and other meta properties
+   * @return {void}
+   */
+  _watchedChange(meta) {
+    console.log('change detected:', meta);
+    this._registeredWatchers.forEach(w => {
+      if(w.collection === 'all' || w.collection === meta.collection) {
+        w.callback(meta);
+      }
+    });
+
+  },
 
+  _changesEmberData(meta) {
```

Comment:
this is really rough and ready but illustrates the general approach I was thinking